tools/castle-editor/code/framedesign.pas

```diff
@@ -3033,28 +3033,10 @@ procedure TDesignFrame.SetParent(AParent: TWinControl);
 function TDesignFrame.ProposeName(const ComponentClass: TComponentClass;
   const ComponentsOwner: TComponent): String;
 var
-  ResultBase: String;
   I: Integer;
+  ResultBase: String;
 begin
-  ResultBase := ComponentClass.ClassName;
-
-  // remove common prefixes
-  if IsPrefix('TCastleUserInterface', ResultBase, true) then
-    ResultBase := PrefixRemove('TCastleUserInterface', ResultBase, true)
-  else
-  if IsPrefix('TCastle', ResultBase, true) then
-    ResultBase := PrefixRemove('TCastle', ResultBase, true)
-  else
-  if IsPrefix('T', ResultBase, true) then
-    ResultBase := PrefixRemove('T', ResultBase, true);
-
-  // remove 2D, as component name cannot start with that
-  if IsPrefix('2D', ResultBase, true) then
-    ResultBase := PrefixRemove('2D', ResultBase, true);
-
-  // in case the replacements above made '', fix it (can happen in case of TCastleUserInterface)
-  if ResultBase = '' then
-    ResultBase := 'Group';
+  ResultBase := FindRegisteredComponentByComponentClass(ComponentClass).ComponentNameBase;
```

Comment:
This makes the `ProposedName`, and so also `AddComponent` that calls it, work only with components that are on the `RegisteredComponents` list.

While it is indeed right now the only use-case, but 

1. If we decide to go this way, we could make it simpler, and `ProposedName` and `AddComponent` could be called with just a `const R: TRegisteredComponent` parameter. We have this information, right now `AddComponent` is called like this:

```
procedure TDesignFrame.MenuItemAddComponentClick(Sender: TObject);
var
  R: TRegisteredComponent;
begin
  R := TRegisteredComponent(Pointer((Sender as TComponent).Tag));
  AddComponent(R.ComponentClass, R.OnCreate);
end;
```

   So it "deconstructs" the `TRegisteredComponent`, while your code would make it connected again.

2. But I *do not* think we should go this way. Better make `ProposedName`, and so also `AddComponent`, work with any class, not necessarily registered one -- for the future.

3. Also, `FindRegisteredComponentByComponentClass` is not really something we can implement. Some classes are present multiple times, just with different `TRegisteredComponent.OnCreate` value (see how `castlescene.pas` registers components like `Scene (Primitive: Box)`).

4. Points 1-3 mean that we should take `const R: TRegisteredComponent` parameter in `AddComponent` and `ProposeName`, and accept it may be `nil` also, and just use the base `ClassName` if `R` is `nil`. I say that this is too unnecessary complication. Let's resign from it, and do not modify `TRegisteredComponent` for it.

Reply:
Sure! If there is a more simple way to - great :)